docs/src/examples/basics.md

```diff
@@ -0,0 +1,141 @@
+Note that the order of the arguments is ``w+xi+yj+zk``, not ``xi+yj+zk+w``.
+
+```@repl intro
+using Quaternions
+q1 = Quaternion(1,2,3,4)
+q2 = quat(5,6,7,8)
+q3 = quat(9)
+```
+
+The multiplication is not commutative.
+```@repl intro
+q1 * q2
+q2 * q1
+```
+
+The multiplicative inverse can be calculated with [`Base.inv`](@ref).
+```@repl intro
+inv(q1)
+inv(q1) * q1
+```
+
+The division is also not commutative.
+
+```@repl intro
+q1 / q2  # Same as `q1*inv(q2)` mathematically.
+q2 \ q1  # Same as `inv(q2)*q1` mathematically.
+```
+
+A conjugate of a quaternion can be calculated with [`Base.conj`](@ref).
+But `Base.imag(::Quaternion)` is not defined because it should return three real values which is not consistent with `imag(::Complex)` and `imag(::Real)`.
+Instead, [`imag_part`](@ref) function can be used to obtain the imaginary part of a quaternion.
+See [issue#61](https://github.com/JuliaGeometry/Quaternions.jl/issues/61) for more discussion.
+
+```@repl intro
+conj(q1)
+imag(q1)  # Not supported.
+imag_part(q1)  # Use this instead.
+```
+
+## `Quaternion` vs `quat`
+The general rule is that [`quat`](@ref) is to [`Quaternion`](@ref) as [`complex`](https://docs.julialang.org/en/v1/base/numbers/#Base.complex-Tuple{Complex}) is to [`Complex`](https://docs.julialang.org/en/v1/base/numbers/#Base.Complex).
+`Complex` and `Quaternion` are both constructors so should return an object of the corresponding type, whereas `quat` and `complex` both can operate on types and arrays.
+
+```@setup Quaternion-quat
+using Quaternions
+```
+
+```@repl Quaternion-quat
+Quaternion(1,2,3,4)
+quat(1,2,3,4)
+Quaternion(Int)  # Similar to `Complex(Int)`.
+quat(Int)  # Similar to `complex(Int)`.
+```
+
+## The type parameter `T` in `Quaternion{T}`
+
+The type parameter `T <: Real` in `Quaternion{T}` represents the type of real and imaginary parts of a quaternion.
+
+### Lipschitz quaternions
+By using this type parameter, some special quaternions such as [**Lipschitz quaternions**](https://en.wikipedia.org/wiki/Hurwitz_quaternion) ``L`` can be represented.
+
+```math
+L = \left\{a+bi+cj+dk \in \mathbb{H} \mid a,b,c,d \in \mathbb{Z}\right\}
+```
+
+```@setup LipschitzHurwitz
+using Quaternions
+```
+
+```@repl LipschitzHurwitz
+q1 = Quaternion{Int}(1,2,3,4)
+q2 = Quaternion{Int}(5,6,7,8)
+islipschitz(q::Quaternion) = isinteger(q.s) & isinteger(q.v1) & isinteger(q.v2) & isinteger(q.v3)
+islipschitz(q1)
+islipschitz(q2)
+islipschitz(q1 + q2)
+islipschitz(q1 * q2)
+islipschitz(q1 / q2)  # Division is not defined on L.
+q1 * q2 == q2 * q1  # non-commucative
+```
+
+### Hurwitz quaternions
+If all coefficients of a quaternion are integers or half-integers, the quaternion is called [**Hurwitz quaternion**](https://en.wikipedia.org/wiki/Hurwitz_quaternion).
+The set of Hurwitz quaternions is defined by
+
+```math
+H = \left\{a+bi+cj+dk \in \mathbb{H} \mid a,b,c,d \in \mathbb{Z} \ \text{or} \ a,b,c,d \in \mathbb{Z} + \tfrac{1}{2}\right\}.
```

Comment:
```suggestion
H = \left\{a+bi+cj+dk \in \mathbb{H} \mid 2a,2b,2c,2d \in \mathbb{Z}\}.
```

Reply:
I was also confused at first, but $1+i/2+j+k$ is **NOT** a Hurwitz quaternion. coefficients should be all integers or all half-integers.

[Hurwitz quaternion (Wikipedia)](https://en.wikipedia.org/wiki/Hurwitz_quaternion)

![image](https://github.com/JuliaGeometry/Quaternions.jl/assets/7488140/48529bb9-7cd8-4cc0-9ff5-62771c522159)

[Factorization of Hurwitz Quaternions](https://citeseerx.ist.psu.edu/viewdoc/download;jsessionid=A8128E37E096B2CEB47746ABABA46DB7?doi=10.1.1.675.7314&rep=rep1&type=pdf)

![image](https://github.com/JuliaGeometry/Quaternions.jl/assets/7488140/3f34b610-9e91-404d-a6d1-bb0cda824ec7)

Reply:
Ah, okay, interesting.